orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTask.java

```diff
@@ -54,22 +66,66 @@ public long getBackoffPeriod() {
 
   @Override
   public long getTimeout() {
-    // TODO(mvulfson): Use DeploymentMonitorDefinition
-    return TimeUnit.MINUTES.toMillis(30);
+    // NOTE: This is not used since we override getDynamicTimeout
+    return 0;
```

Comment:
If it's not meant to be used, shouldn't it throw an exception instead? Others might miss your comment here...

Reply:
yeah, this is a bit weird. maybe we should just get rid of this method (but that's a bit more work that i don't want to lump here).
If you look at `RunTaskHandler` (https://github.com/spinnaker/orca/blob/725db2b0cadd6262ba6c8bafb4888f57fcf7f358/orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/RunTaskHandler.kt#L241) it always calls the `getDynamicTimeout`. and the only place where `getTimeout` is called by https://github.com/spinnaker/orca/blob/725db2b0cadd6262ba6c8bafb4888f57fcf7f358/orca-core/src/main/java/com/netflix/spinnaker/orca/RetryableTask.java#L30 but since i override it it will never be called

Could add an exception, but i would rather just delete this function in the medium term